### PR TITLE
GG-21171

### PR DESCRIPTION
**Pull request by antonovsergey93**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java [hunk 1]
```diff
@@ -1517,20 +1518,40 @@ private long configuredOffheap() {
      */
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java:1518]: Please add a new parameter to JavaDoc.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java [hunk 2]
```diff
@@ -1517,20 +1518,40 @@ private long configuredOffheap() {
...
     private void topologySnapshotMessage(IgniteClosure<String, Void> clo, long topVer, DiscoCache discoCache,
         int evtType, ClusterNode evtNode, int srvNodesNum, int clientNodesNum, int totalCpus, double heap,
-        double offheap) {
+        double offheap, boolean needNodesDetails) {
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java:1521]: The method's signature violates our code style. Please place each parameter on a separate line. Look at {{GridDiscoveryManager#fillAffinityNodeCaches}} method as a reference.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java [hunk 3]
```diff
@@ -1517,20 +1518,40 @@ private long configuredOffheap() {
...
+            summary.a(", nodes=[");
+
+            String prefix = "";
+
+            for (ClusterNode clusterNode : discoCache.allNodes()) {
+                summary.a(prefix);
+
+                prefix = ",";
+
+                summary.a(clusterNode.toString());
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java:1546]: I'd prefer always add a comma after each node and remove last comma by `summary.setLength(summary.length() - 1);`

File: modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java [hunk 4]
```diff
@@ -1517,20 +1518,40 @@ private long configuredOffheap() {
...
+        summary.a(", CPUs=").a(totalCpus);
+        summary.a(", offheap=").a(offheap).a("GB");
+        summary.a(", heap=").a(heap).a("GB");
+
+        if (evtType == EVT_NODE_JOINED && needNodesDetails) {
+            summary.a(", nodes=[");
+
+            String prefix = "";
+
+            for (ClusterNode clusterNode : discoCache.allNodes()) {
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java:1541]: I think we should print only alive nodes. You can check it by `DiscoCache#alive`. Don't forget to update prefix `", nodes=["` too.

File: modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java [hunk 5]
```diff
@@ -238,18 +242,40 @@ private void doForceServerAndClientTest(MockLogger log) throws Exception {
...
+    }
+
+    /**
+     * Check nodes details in log.
+     *
+     * @param log log.
+     */
+    private void checkNodesAdditionalLogging(MockLogger log){
+        if (log.isDebugEnabled())
+            assertFalse(F.exist(log.logs(), new IgnitePredicate<String>() {
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java:256]: can be simplified to `assertFalse(F.exist(log.logs(), s -> s.contains("nodes=[")));`

File: modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java [hunk 6]
```diff
@@ -105,15 +105,15 @@ private void doServerLogTest(MockLogger log) throws Exception {
         String nodeId8;
 
         try {
-            Ignite server = startGrid("server");
```
Comment (antonovsergey93) [inline modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java:108]: I think we can keep variables names as-is.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java (hunk 2 above)
Comment (makedonskaya) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java:1521]: I read Coding Guidelines. There are two styles of method declaration.
https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines
If arguments can't fit to single line, arguments are splitted by 120 char limit to several lines.

File: modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java [hunk 7]
```diff
@@ -37,17 +32,24 @@
...
-    public static final String TOPOLOGY_SNAPSHOT = "Topology snapshot";
+    private static final String ALIVE_NODES_MSG = "aliveNodes=[";
 
     /** */
-    public static final String SERV_NODE = ">>> Number of server nodes";
+    private static final String NUMBER_SRV_NODES = ">>> Number of server nodes: %d";
 
     /** */
-    public static final String CLIENT_NODE = ">>> Number of client nodes";
+    private static final String NIMBER_CLIENT_NODES = ">>> Number of client nodes: %d";
```
Comment (sanpwc) [inline modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java:41]: Typo?

File: modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java [hunk 8]
```diff
@@ -62,237 +64,209 @@
...
+     * @param dbg Debug flag.
      * @throws Exception If failed.
      */
-    private void doServerLogTest(MockLogger log) throws Exception {
+    private void doServerLogTest(boolean dbg) throws Exception {
         String nodeId8;
 
+        testLog = new ListeningTestLogger(dbg, log);
+
+        LogListener aliveNodesLsnr = LogListener.matches(ALIVE_NODES_MSG).times(dbg ? 0 : 4).build();
```
Comment (sanpwc) [inline modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java:106]: Is that enough to check that "aliveNodes=["  is present in logs, I believe that content also matters here.

File: modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java [hunk 9]
```diff
@@ -62,237 +64,209 @@
...
-        if (log instanceof MockLogger)
-            ((MockLogger)log).clear();
+        super.afterTest();
     }
 
     /**
      * @throws Exception If failed.
      */
     @Test
     public void testServerLogs() throws Exception {
```
Comment (sanpwc) [inline modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java:83]: Here and below. What about adding few words within test comment about what test is supposed to do?

**Pull request closed without merging**